crates/miden-lib/src/transaction/memory.rs

```diff
@@ -288,13 +288,14 @@ pub const NOTE_MEM_SIZE: MemoryAddress = 2048;
 // Each nullifier occupies a single word. A data section for each note consists of exactly 512
 // words and is laid out like so:
 //
-// ┌──────┬────────┬────────┬────────┬────────┬──────┬───────┬────────┬───────┬─────┬───────┬─────────┬
-// │ NOTE │ SERIAL │ SCRIPT │ INPUTS │ ASSETS │ META │ NOTE  │   NUM  │ ASSET │ ... │ ASSET │ PADDING │
-// │  ID  │  NUM   │  ROOT  │  HASH  │  HASH  │ DATA │ ARGS  │ ASSETS │   0   │     │   n   │         │
-// ├──────┼────────┼────────┼────────┼────────┼──────┼───────┼────────┼───────┼─────┼───────┼─────────┤
-// 0      4        8        12       16       20     24      28       32 + 4n
+// ┌──────┬────────┬────────┬────────┬────────┬──────┬───────┬────────┬────────┬───────┬─────┬───────┬─────────┬
+// │ NOTE │ SERIAL │ SCRIPT │ INPUTS │ ASSETS │ META │ NOTE  │  NUM   │  NUM   │ ASSET │ ... │ ASSET │ PADDING │
+// │  ID  │  NUM   │  ROOT  │  HASH  │  HASH  │ DATA │ ARGS  │ INPUTS │ ASSETS │   0   │     │   n   │         │
+// ├──────┼────────┼────────┼────────┼────────┼──────┼───────┼────────┼────────┼───────┼─────┼───────┼─────────┤
+// 0      4        8        12       16       20     24      28       32       36 + 4n
 //
-// - NUM_ASSETS is encoded [num_assets, 0, 0, 0].
+// - NUM_INPUTS is encoded as [num_inputs, 0, 0, 0].
+// - NUM_ASSETS is encoded as [num_assets, 0, 0, 0].
```

Comment:
What's the reason for using two words here? Could this not be:
```
[num_inputs, num_assets, 0, 0]
```
But also, if having two words simplifies things, it's fine to keep.

Reply:
I thought about it, but decided that it could be confusing. But if we decide to move the note inputs from the advice provider to the memory, that would make more sense: each word would contain the number of corresponding values.